nautobot/dcim/migrations/0044_ensure_virtual_chassis_names_are_unique.py

```diff
@@ -0,0 +1,34 @@
+# Generated by Django 3.2.18 on 2023-06-06 21:12
+
+from django.db import migrations
+from django.core.exceptions import ValidationError
+
+
+def ensure_virtual_chassis_names_are_all_unique(apps, schema_editor):
+    VC_NAME_TO_COUNT = {}
+    VC = apps.get_model("dcim", "VirtualChassis")
+
+    vcs = VC.objects.all()
+    duplicates = []
+    for vc in vcs:
+        count = VC_NAME_TO_COUNT.setdefault(vc.name, 0)
+        VC_NAME_TO_COUNT[vc.name] = count + 1
+        if count + 1 > 1:
+            duplicates.append(vc.name)
+
+    if duplicates:
+        for duplicate_vc in duplicates:
+            print(f">>> There are {VC_NAME_TO_COUNT[duplicate_vc]} with name {duplicate_vc}\n")
+        print(
+            ">>> Please rename all Virtual Chassis instances with duplicate names before running this migration again.\n"
+        )
+        print(">>> Failing migration...\n")
+        raise ValidationError("Duplicate Virtual Chassis names detected.")
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("dcim", "0043_status_nonnullable"),
+    ]
+
+    operations = []
```

Comment:
I don't think this actually runs the check then. Also, can we have these checks within the same migration where we alter the field? We're not doing any data changes and they should just be performed before the schema migration operation.